### ISSUE TITLE
App service doesn't start due to ambigous network, even though this worked before

### ISSUE THREAD

**Issue by SbiCA** [Azure/appservice-linux#1]
We are using app service linux + docker containers with docker-compose, which seems to work in general. 

Lately we've run into the following issue

```
2019-07-24 14:08:52.090 ERROR - Container start failed for agileplannerbyt-appservice_traefik_0 with System.AggregateException, One or more errors occurred.
InnerException: Docker.DotNet.DockerApiException, Docker API responded with status code=BadRequest, response={"message":"network agileplannerbyt-appservice_multi_nw__0 is ambiguous (1824 matches found on name)"}
```

which we were only able to overcome be creating a completely new app service plan + app service. Note sure why this would not work after some days, since the exact same configuration was working before

**Comment by andreylucasantosss** [comment]
ERROR - Container start failed for insightfy_redis_0 with System.AggregateException, One or more errors occurred. (Docker API responded with status code=BadRequest, response={"message":"network insightfy_multi_nw__0 is ambiguous (329 matches found on name)"}
) (Docker API responded with status code=BadRequest, response={"message":"network insightfy_multi_nw__0 is ambiguous (329 matches found on name)"}

any other solution?

**Comment by skryshtafovych** [comment]
Running into Same issue i guess the previous container images are not cleaned up inside azure or we can see other people images in the Azure subscription 


Container start failed for dockercomposeappletgi_influxdb_0 with System.AggregateException, One or more errors occurred. (Docker API responded with status code=BadRequest, response={"message":"network dockercomposeappletgi_multi_nw__0 is ambiguous (20 matches found on name)"}
) (Docker API responded with status code=BadRequest, response={"message":"network dockercomposeappletgi_multi_nw__0 is ambiguous (20 matches found on name)"}

**Comment by skryshtafovych** [comment]
**According to Microsoft:**


This is a known issue and Product Group is currently investigating.
As a workaround please try the following which has worked for other customers:
In container settings, select single container, select any image from the registry and click save.
Once the Save is success, select Docker Compose on the same page and click Save again.
Wait for some time to check if the issue is resolved by browsing the site.


**Step Below is What Fixed it For My Config**
===============================
If the above steps do not resolve the issue, please perform a scale up/down operation from the portal:
Scale up the App service plan to S1n or p1V2, click Apply and once scaleup is done, check if the site is up.
Once site is up, you can scale back down to original tier.

**Comment by SbiCA** [comment]
@skryshtafovych  Scale up seems to help 😃 thanks a lot for the hint, but I would still expect this to work without scaling up otherwise this feature is pretty much useless IMHO. Because you can't rely on the app service starting up correctly and it would require manual interaction 😢

**Comment by SbiCA** [comment]
Bad news, even with P1V2 or S1 I'm getting the same error

This was with P1V2
```
2019-08-07 10:23:26.778 ERROR - Container start failed for demo-agileplanner_traefik_0 with System.AggregateException, One or more errors occurred.
InnerException: Docker.DotNet.DockerApiException, Docker API responded with status code=BadRequest, response={"message":"network demo-agileplanner_multi_nw__0 is ambiguous (4 matches found on name)"}
```

**Comment by skryshtafovych** [comment]
@SbiCA Agree...  I had to do that multiple times about 3 since it did not work for me the first time. Also I did both steps they suggested scaling and changing container configs. From my understanding they are working on fix and are aware of these open Bugs. My compose container set works fine now with restarts and Cold Starts, but i have not pushed new containers since getting service back up ill update my experience with our next  container release.

**Comment by bigal91** [comment]
Having the same issue here. Is there any progress from the Azure "Product Group" ? The workaround is really hacky.

**Comment by tiago156** [comment]
Same problem here, any news?

**Comment by f1xxxer** [comment]
I am having the same problem about ambiguous network, when using user assigned managed identity with app configuration service and key vault.

**Comment by jcarloslr10** [comment]
@microsoftopensource @msftgits any news about it? I think it's an important enough issue for application stability when choosing AppService with Linux and integration via Managed Identity (which is very common).

I am experiencing this problem a few times randomly. It usually happens every X number of deployments and the only solution is to restart the AppService until it finishes working, but this is very serious when it happens in production environments.

It's not feasible to be scaling up and down the AppServices in production environments. It's a pretty bad smelling solution...

It would be nice to know how you are approaching this issue.

Thanks in advance!

**Comment by Wmaarts** [comment]
I'm glad there's a workaround at all, but this we would really like some news on this topic. With this behaviour as it is, are Azure App Services not yet ready for use in production and did we miss something? 🤔
